pkg/sentry/fsimpl/fuse/fusefs.go

```diff
@@ -227,3 +227,81 @@ func (i *Inode) Open(ctx context.Context, rp *vfs.ResolvingPath, vfsd *vfs.Dentr
 	}
 	return fd.VFSFileDescription(), nil
 }
+
+// Stat implements kernfs.Inode.Stat.
+func (i *Inode) Stat(ctx context.Context, fs *vfs.Filesystem, opts vfs.StatOptions) (linux.Statx, error) {
+	conn := fs.Impl().(*filesystem).conn
+	task, creds := kernel.TaskFromContext(ctx), auth.CredentialsFromContext(ctx)
+	if task == nil {
+		panic("Stat must be called from a task context")
+	}
+
+	var in linux.FUSEGetAttrIn
+	// We don't set any attribute in the request, becuase in VFS2 fstat(2) will
+	// finally be translated into vfs.FilesystemImpl.StatAt() (see
+	// pkg/sentry/syscalls/linux/vfs2/stat.go), resulting in the same flow
+	// as stat(2). Thus GetAttrFlags and Fh variable are currently not used.
+	req, err := conn.NewRequest(creds, uint32(task.ThreadID()), i.Ino(), linux.FUSE_GETATTR, &in)
+	if err != nil {
+		return linux.Statx{}, nil
+	}
+
+	res, err := conn.Call(task, req)
+	if err != nil {
+		return linux.Statx{}, err
+	}
+	if err := res.Error(); err != nil {
+		return linux.Statx{}, err
+	}
+
+	var out linux.FUSEGetAttrOut
+	if err := res.UnmarshalPayload(&out); err != nil {
+		return linux.Statx{}, err
+	}
+
+	var stat linux.Statx
+	stat.Blksize = out.Attr.BlkSize
```

Comment:
[nit] for readability sake, the rest of this function can probably be a call to a helper function to translate a `FUSEGetAttrOut` to an equivalent `linux.Statx` struct.

Reply:
Good idea. Extract to another function.